### PR TITLE
W-15008392: Don't stringify `null` values (this allows us to handle them correctly

### PR DESCRIPTION
**Pull request by iglosiggio**
This change improves handling of `null` values inside arrays and query strings.

### CODE REVIEW

File: interface-impl-amf/src/test/java/org/mule/amf/impl/model/ParameterImplTest.java [hunk 1]
```diff
@@ -195,7 +203,6 @@ public void validateSpecialCharactersInArray() {
     validateArrayValue(nullableArray, nonNullableArray, asList("Tests:Tests"));
     validateArrayValue(nullableArray, nonNullableArray, asList("Test:%20Test"));
     validateArrayValue(nullableArray, nonNullableArray, asList("Test%3A%20Test"));
-    validateArrayValue(nullableArray, nonNullableArray, asList("*qwerty*qwerty*", null));
```
Comment (iglosiggio) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/model/ParameterImplTest.java:198]: This test was invalid. `nullableArray` is an array that may be just `null` but its values **should** be strings!

File: interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java [hunk 2]
```diff
@@ -121,6 +123,46 @@ public void multipleUnionsValidationTest() {
...
+        put("nullableString", singletonList("null"));
+        put("nonNullableString", singletonList("null"));
+      }
+    }));
+  }
+
+  @Test
+  public void nullableParameterQueryString() {
+
+  }
```
Comment (santiagoGQ) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java:164]: Is the test blank on purpouse?

**Review by santiagoGQ — APPROVED** [review]

File: interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java (hunk 2 above)
Comment (iglosiggio) [reply interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java:164]: wooooops, it looks like a typo 😅 
```suggestion
```

File: interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java [hunk 3]
```diff
@@ -121,6 +123,42 @@ public void multipleUnionsValidationTest() {
...
+    assertTrue(nullableQueryString.validate(new HashMap<String, Collection<?>>() {
+
+      {
+        put("nullableString", singletonList("null"));
+        put("nonNullableString", singletonList("null"));
+      }
+    }));
+  }
+
+
```
Comment (iglosiggio) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/model/QueryStringImplTest.java:161]:
```suggestion
```

**Review by santiagoGQ — APPROVED** [review]

**Pull request merged**